.github/workflows/lint_workflows.yml

```
@@ -0,0 +1,19 @@
+name: "Lint Workflows"
+
+on:
+  pull_request:
+    branches: [main]
+permissions:
+  contents: read
+jobs:
+  actionlint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download actionlint
+        id: get_actionlint
+        run: bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash)
```

This is a bit of a security hole... if the actionlint repo is compromised and something bad pushed to the main branch, we'll immediately let it into our CI.

Ideally we wouldn't use the download script, it's strange that that's the recommended way of using actionlint in CI. At the very least let's change the URL to a pinned shasum:

```suggestion
        run: bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/03d0035246f3e81f36aed592ffb4bebf33a03106/scripts/download-actionlint.bash)
```

---

Personally I would advocate for copying the actually useful bits out of https://github.com/rhysd/actionlint/blob/main/scripts/download-actionlint.bash into our action. We don't need most of what that script does, we can download the executable from GitHub releases and make it executable ourselves.